modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java

```diff
@@ -58,43 +56,39 @@
    * @param name Name of the session factory.
    */
   public TransactionalRequest(@Nonnull String name) {
+    sessionRequest = new SessionRequest(name);
     key = ServiceKey.key(SessionFactory.class, name);
   }
 
   /**
    * Creates a new transactional request and attach to the default/first session factory registered.
    */
   public TransactionalRequest() {
+    sessionRequest = new SessionRequest();
     key = ServiceKey.key(SessionFactory.class);
   }
 
   @Nonnull @Override public Route.Handler apply(@Nonnull Route.Handler next) {
-    return ctx -> {
+    return sessionRequest.apply(ctx -> {
       SessionFactory sessionFactory = ctx.require(key);
       Transaction trx = null;
       try {
-        Session session = sessionFactory.openSession();
-        ManagedSessionContext.bind(session);
+        Session session = sessionFactory.getCurrentSession();
         trx = session.getTransaction();
-        trx.begin();
```

Comment:
seems wrong, why did you remove the begin trx?

Reply:
I meant to change this to session.beginTransaction(); Which creates a transaction, begins and returns that tx.